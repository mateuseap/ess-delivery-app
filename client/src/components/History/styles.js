import styled from "styled-components";

import { StyledButton } from "../../utils/styles";
import { DEFAULT_RED, LIGHT_RED, DARK_RED } from "../App/App.style";

export const MainButton = styled(StyledButton)`
  width: 235px;
  font-size: 20px;
  white-space: nowrap;
  padding: 12px 6% 12px 6%;
  display: inline-block;
  font-weight: normal;
`;

export const ImageStyle = styled.img`
  width: 234px;
  height: 170px;
  border-radius: 65px;
  content: url(${(props) => props.photoUrl});
`;

export const DisabledSendButton = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

export const DescriptionStyle = styled.div`
  display: grid;
  width: 200px;
  grid-auto-flow: row;
  grid-column-gap: 4vw;
`;

export const ActionButtonsStyle = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const RateLabel = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 5em;
  min-width: 30vw;
  font-size: 40px;
  line-height: 47px;
`;

export const MainDiv = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 10vw;
`;

export const TableDataStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const PageStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const BorderText = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 22vw;
  color: white;
  padding: 0px 10px 0px 10px;
  text-align: center;
  background: ${LIGHT_RED};
  border-radius: 63px;
  font-size: 2rem;
  line-height: 1.5rem;
`;

export const RectangleFilter = styled.div`
  width: 230px;
  height: 60px;
  display: flex;
  gap: 8%;
  padding: 12px;

  color: white;
  font-weight: 400;
  font-size: 16px;

  justify-content: center;
  align-items: center;

  background: ${DEFAULT_RED};
  border-radius: 100px;
`;

export const RectangleDaysFilter = styled(RectangleFilter)`
  width: 80px;
  height: 40px;
  padding: 0;

  background: ${LIGHT_RED};
  border-radius: 63px;
`;

export const SelectStyle = styled.select`
  padding: 0 1px;
  background: transparent;
  cursor: pointer;
  line-height: 1;
  border: 0;
  color: white;
  border-radius: 63px;
  text-align: center;
  height: 100%;
  width: 100%;
  -webkit-appearance: none;
`;

export const TopDiv = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 10vw;
`;

export const CirclesStyle = styled(StyledButton)`
  padding: 0;
  width: 40px;
  font-size: 16px;
  height: 40px;
  display: inline-block;
  background: ${LIGHT_RED};
  border-radius: 50%;
  &:hover {
    border: 1px solid yellow;
    background-color: ${DARK_RED};
  }
`;

export const NoDataStyle = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  top: 45%;
  left: 0;
  right: 0;
  margin: auto;
`;

export const OptionStyle = styled.option`
  color: black;
`;
