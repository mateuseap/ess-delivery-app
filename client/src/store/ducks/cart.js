import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  cartRequest: [],
  getCart: [],
  updateCart: ["rest_id", "rest_name", "item", "amountToChange"],
  makeOrder: ["callback"],
  cartSuccess: ["data"],
  cartError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.CART_ERROR]: error,
  [Types.CART_REQUEST]: request,
  [Types.CART_SUCCESS]: success,
});
