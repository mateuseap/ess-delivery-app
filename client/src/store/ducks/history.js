import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  historyRequest: [],
  getHistory: ["dateFilter"],
  postHistory: ["data"],
  historySuccess: ["data"],
  historyError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.HISTORY_ERROR]: error,
  [Types.HISTORY_REQUEST]: request,
  [Types.HISTORY_SUCCESS]: success,
});
