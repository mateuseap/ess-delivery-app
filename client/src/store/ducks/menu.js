import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  menuRequest: [],
  getMenu: ["id"],
  menuSuccess: ["data"],
  menuError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.MENU_ERROR]: error,
  [Types.MENU_REQUEST]: request,
  [Types.MENU_SUCCESS]: success,
});
