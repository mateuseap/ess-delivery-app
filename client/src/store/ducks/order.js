import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  orderRequest: [],
  getOrderDetails: ["id"],
  orderSuccess: ["data"],
  orderError: ["err"],
  cancelOrder: ["id", "callback"],
  cancelOrderWatch: [],
  orderStatusWatchWorker: ["id"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

const cancelOrderWatch = (state) => state.merge({});

export default createReducer(INITIAL_STATE, {
  [Types.ORDER_ERROR]: error,
  [Types.ORDER_REQUEST]: request,
  [Types.ORDER_SUCCESS]: success,
  [Types.CANCEL_ORDER_WATCH]: cancelOrderWatch,
});
