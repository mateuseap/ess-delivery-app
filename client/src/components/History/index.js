import React, { Component } from "react";
import { Link } from "react-router-dom";

import {
  RateLabel,
  BorderText,
  PageStyle,
  MainDiv,
  TableDataStyle,
  ActionButtonsStyle,
  ImageStyle,
  DescriptionStyle,
  DisabledSendButton,
  RectangleFilter,
  RectangleDaysFilter,
  SelectStyle,
  TopDiv,
  NoDataStyle,
  OptionStyle,
  CirclesStyle,
  MainButton,
} from "./styles";

// Pop-up -> detalhes
import Popup from "reactjs-popup";
import "reactjs-popup/dist/index.css";

import ReactLoading from "react-loading";

import { connect } from "react-redux";
import { Creators as HistoryCreator } from "../../store/ducks/history";
import { Table, Button, Form } from "react-bootstrap";
import ReactStars from "react-rating-stars-component";

import { formatMoney } from "../../utils/misc";
import { deliveryFee } from "../../constants/constants";

const ITEMS_PER_PAGE = 2;

class History extends Component {
  state = {
    orderToRate: -1,
    elementToRateId: -1,
    currentStarsValue: 0,
    currentFeedbackText: "",
    changeSelectedBg: -1,
    canSendRate: false, // A avaliação com estrelas é obrigatória
    daysFilter: 30,
    currentPage: 0,
  };

  componentDidMount() {
    this.props.getHistory(this.state.daysFilter);
  }

  clearRate(aditionalStates) {
    this.setState({
      orderToRate: -1,
      elementToRateId: -1,
      currentFeedbackText: "",
      currentStarsValue: 0,
      changeSelectedBg: -1,
      ...aditionalStates,
    });
  }

  cancelRate() {
    if (this.state.currentFeedbackText || this.state.currentStarsValue) {
      const resp = window.confirm(
        "Se você cancelar, todo progresso dessa avaliação será perdido"
      );
      if (resp) this.clearRate();
    } else this.clearRate();
  }

  handleSelectFilter(elem) {
    this.props.getHistory(elem.target.value);
    this.clearRate({
      daysFilter: elem.target.value,
      currentPage: 0,
    });
  }

  handlePageChange(elem) {
    this.clearRate({
      currentPage: elem.target.childNodes[0].data - 1,
    });
  }

  handleRating(index, element) {
    this.setState({
      orderToRate: index,
      elementToRateId: element.id,
      changeSelectedBg: index,
      canSendRate: this.props.history.data[index].rate.stars !== 0,
    });
  }

  sendRate() {
    const {
      orderToRate,
      elementToRateId,
      currentStarsValue,
      currentFeedbackText,
      daysFilter,
    } = this.state;

    this.props.postHistory({
      restaurantId: this.props.history.data[orderToRate].restaurant_id,
      rate: {
        stars: currentStarsValue,
        feedback_text: currentFeedbackText,
      },
      orderId: elementToRateId,
      daysFilter: daysFilter,
    });
  }

  getCircles() {
    const numberOfCircles = Math.ceil(
      this.props.history.data.length / ITEMS_PER_PAGE
    );

    return Array.from(Array(numberOfCircles).keys());
  }

  handleStars(newRating) {
    this.setState({
      currentStarsValue: newRating,
      canSendRate: newRating !== 0,
    });
  }

  renderDaysFilter() {
    return (
      <RectangleFilter>
        Filtro de Dias
        <RectangleDaysFilter>
          <SelectStyle 
            name="daysFilter" 
            value={this.state.daysFilter}
            onChange={(elem) => {
              this.handleSelectFilter(elem);
            }}
          >
            <OptionStyle value="30">30 dias</OptionStyle>
            <OptionStyle value="15">15 dias</OptionStyle>
            <OptionStyle value="7">7 dias</OptionStyle>
          </SelectStyle>
        </RectangleDaysFilter>
      </RectangleFilter>
    );
  }

  renderOrderTableRow(element, index) {
    const { orderToRate, changeSelectedBg } = this.state;
    return (
      <tr
        key={element.id}
        style={
          changeSelectedBg === index
            ? {
                backgroundColor: "rgba(0, 0, 0, 0.2)",
              }
            : {}
        }
      >
        <ImageStyle photoUrl={element.orderImage} />
        <td className="p-2">
          <TableDataStyle>
            <p style={{ fontSize: 26 }}>{element.restaurant_name}</p>
            <p>{element.description[0].name}</p>
            <p style={{ fontWeight: "bold" }}>
              {formatMoney(element.total_price + deliveryFee)}
            </p>
          </TableDataStyle>
        </td>
        <td>
          {!element.status.finished ? (
            <Link to={`/details/${element.id}`}>
              <MainButton variant="green">Acompanhar pedido</MainButton>
            </Link>
          ) : (
            <MainButton
              name="ratingButton"
              variant={element.rate.did ? "" : "blue"}
              disabled={orderToRate >= 0}
              onClick={() => this.handleRating(index, element)}
            >
              {element.rate.did ? "Revisar avaliação" : "Avaliar Pedido"}
            </MainButton>
          )}
        </td>
      </tr>
    );
  }

  renderPagination() {
    return (
      <RectangleFilter className="mt-3">
        {this.getCircles().map((element, index) => (
          <CirclesStyle
            name="pagination"
            key={index}
            style={
              this.state.currentPage === element
                ? { border: "1px solid #ffffff" }
                : {}
            }
            onClick={(elem) => this.handlePageChange(elem)}
          >
            {element + 1}
          </CirclesStyle>
        ))}
      </RectangleFilter>
    );
  }

  render() {
    const { orderToRate, canSendRate, currentPage } = this.state;
    const { history } = this.props;
    const { data, loading } = history;

    return (
      <PageStyle>
        {loading ? (
          <ReactLoading
            type={"spin"}
            style={{
              position: "absolute",
              width: "10vw",
              top: "40%",
              left: "0",
              right: "0",
              margin: "auto",
            }}
          />
        ) : (
          <>
            {data?.length ? (
              <>
                <TopDiv className="m-3">
                  <BorderText>Histórico de Pedidos</BorderText>
                  {this.renderDaysFilter()}
                </TopDiv>
                <MainDiv>
                  <Table borderless>
                    <tbody>
                      {data
                        .slice(
                          currentPage * ITEMS_PER_PAGE,
                          (currentPage + 1) * ITEMS_PER_PAGE
                        )
                        .map((element, index) => {
                          return this.renderOrderTableRow(element, index);
                        })}
                    </tbody>
                  </Table>
                  {orderToRate > -1 ? (
                    <Form style={{ maxWidth: 650 }}>
                      <Form.Group controlId="userFeedback">
                        <Form.Label
                          style={{
                            display: "flex",
                            alignItems: "center",
                            flexDirection: "column",
                          }}
                        >
                          <RateLabel>
                            <h2>{data[orderToRate].description[0].name}</h2>
                            <Popup
                              trigger={
                                <Button variant="warning">Detalhes</Button>
                              }
                              position="left center"
                            >
                              <DescriptionStyle className="p-2">
                                {data[orderToRate].description?.map(
                                  (element) => (
                                    <div key={element.name}>
                                      <h5>{element.name}</h5>
                                      <p>
                                        {formatMoney(
                                          element.price * element.quantity
                                        )}
                                      </p>
                                    </div>
                                  )
                                )}
                                <strong>Taxa de entrega</strong>
                                <p>{formatMoney(deliveryFee)}</p>
                                <div style={{ color: "#630606" }}>
                                  <strong>Forma de pagamento</strong>
                                  <p>{data[orderToRate].payment.type}</p>
                                </div>
                              </DescriptionStyle>
                            </Popup>
                          </RateLabel>
                          <h3 className="mt-4">
                            {formatMoney(
                              data[orderToRate].total_price + deliveryFee
                            )}
                          </h3>
                          <ReactStars
                            count={5}
                            onChange={this.handleStars.bind(this)}
                            isHalf={true}
                            value={data[orderToRate].rate.stars}
                            edit={!data[orderToRate].rate.did}
                            size={50}
                            activeColor="#ffd700"
                          />
                        </Form.Label>
                        <Form.Control
                          disabled={data[orderToRate].rate.did}
                          name="ratingFeedbackText"
                          as="textarea"
                          type="text"
                          className="mr-2"
                          defaultValue={data[orderToRate].rate.feedback_text}
                          rows={12}
                          placeholder={
                            !data[orderToRate].rate.did
                              ? "Deixe seu feeback!"
                              : ""
                          }
                          onChange={(text) =>
                            this.setState({
                              currentFeedbackText: text.target.value,
                            })
                          }
                          style={{ fontWeight: "600" }}
                        />
                        <Form.Text className="text-muted">
                          Sua avaliação nos ajuda a melhorar a experiência do
                          app :)
                        </Form.Text>
                      </Form.Group>
                      {!data[orderToRate].rate.did ? (
                        <>
                          <ActionButtonsStyle className="mt-2">
                            <Button
                              variant="secondary"
                              name="cancelRateButton"
                              onClick={() => this.cancelRate()}
                            >
                              Cancelar
                            </Button>
                            <Button
                              variant="danger"
                              disabled={!canSendRate}
                              name="sendRateButton"
                              onClick={() => this.sendRate()}
                            >
                              Enviar
                            </Button>
                          </ActionButtonsStyle>
                          {!canSendRate ? (
                            <DisabledSendButton>
                              <label>
                                Avaliação com estrelas é obrigatória para
                                continuar!
                              </label>
                            </DisabledSendButton>
                          ) : null}
                        </>
                      ) : (
                        <Button
                          variant="outline-primary"
                          name="backButton"
                          onClick={() => this.clearRate()}
                          className="mt-2"
                        >
                          Voltar
                        </Button>
                      )}
                    </Form>
                  ) : null}
                </MainDiv>
                {this.renderPagination()}
              </>
            ) : (
              <NoDataStyle>
                <h1 name="emptyOrdersHistory">Não há pedidos registrados em sua conta</h1>
                <h2>Volte a página inicial</h2>
                <Link to="/home" className="m-2">
                  <Button>Voltar</Button>
                </Link>
              </NoDataStyle>
            )}
          </>
        )}
      </PageStyle>
    );
  }
}

const mapStateToProps = ({ history }) => ({
  history,
});

export default connect(mapStateToProps, {
  ...HistoryCreator,
})(History);
