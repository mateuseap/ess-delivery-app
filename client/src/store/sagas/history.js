import { call, put } from "@redux-saga/core/effects";
import { Creators } from "../ducks/history";
import api from "../../services/api";
import { toastr } from "react-redux-toastr";

export function* getHistory({ dateFilter }) {
  try {
    yield put(Creators.historyRequest());

    const response = yield call(api.get, "/orders", { params: { dateFilter } });

    if (response.data) {
      yield put(Creators.historySuccess(response.data));
    }
  } catch (err) {
    yield put(Creators.historyError({ err }));
    toastr.error("Erro ao pegar histórico dos pedidos");
  }
}

export function* postHistory({ data }) {
  try {
    yield put(Creators.historyRequest());

    const response = yield call(api.post, "/orders", { ...data });

    if (response.data) {
      yield put(Creators.historySuccess(response.data));
      toastr.success("Avaliação enviada com sucesso.");
    }
  } catch (err) {
    yield put(Creators.historyError({ err }));
    toastr.error("Erro ao enviar avaliação.");
  }
}
