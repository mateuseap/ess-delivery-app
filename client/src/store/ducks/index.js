import { combineReducers } from "redux";

import { reducer as toastr } from "react-redux-toastr";

import user from "./user";
import restaurants from "./restaurants";
import history from "./history";
import cart from "./cart";
import menu from "./menu";
import order from "./order";

export default combineReducers({
  user,
  restaurants,
  history,
  cart,
  menu,
  order,
  toastr,
});
