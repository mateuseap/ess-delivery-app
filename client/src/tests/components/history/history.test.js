import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import History from "../../../components/History";
import { BrowserRouter } from "react-router-dom";
import { INITIAL_STATE, Types } from "../../../store/ducks/history";
import createSagaMiddleware from "redux-saga";
import { historyLoadedState } from "../dataUtils";

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

const sagaMiddleware = createSagaMiddleware();

const mockStore = configureStore([sagaMiddleware]);

Enzyme.configure({ adapter: new Adapter() });

describe("Home Component Integration Testing", () => {
  it("Renders loading correctly", () => {
    const store = mockStore({ history: INITIAL_STATE });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <History />
        </BrowserRouter>
      </Provider>
    );

    const expectedActions = [{ type: Types.GET_HISTORY, dateFilter: 30 }];
    expect(store.getActions()).toEqual(expectedActions);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Renders data correctly", () => {
    const store = mockStore({
      history: historyLoadedState,
    });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <History />
        </BrowserRouter>
      </Provider>
    );
    const expectedActions = [{ type: Types.GET_HISTORY, dateFilter: 30 }];
    expect(store.getActions()).toEqual(expectedActions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
