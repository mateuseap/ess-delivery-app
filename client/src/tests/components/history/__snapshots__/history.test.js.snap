// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Integration Testing Renders data correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(History)>
        <History
          getHistory={[Function]}
          history={
            Object {
              "data": Array [
                Object {
                  "date": "2022-04-20",
                  "description": Array [
                    Object {
                      "item_id": 1,
                      "name": "Ogro Burguer com Batata Média",
                      "price": 25.5,
                      "quantity": 1,
                    },
                    Object {
                      "item_id": 2,
                      "name": "Coca-cola lata",
                      "price": 5.5,
                      "quantity": 1,
                    },
                  ],
                  "from": "Avenida Conselheiro Rosa e Silva, 1086",
                  "id": "10bbd312-47d8-4513-894f-a6556620e763",
                  "orderImage": "https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg",
                  "payment": Object {
                    "code": "XXXX-XXXX-XXXX-XXXX",
                    "name": "Meu cartão NuBank",
                    "security_code": "XXX",
                    "type": "Cartão de Débito",
                  },
                  "rate": Object {
                    "did": false,
                    "feedback_text": "",
                    "stars": 0,
                  },
                  "restaurant_id": 2,
                  "restaurant_name": "Elias",
                  "status": Object {
                    "confirmed": true,
                    "delivering": true,
                    "finished": true,
                    "preparing": true,
                  },
                  "to": "Avenida Jornalista Aníbal Fernandes, 500",
                  "total_price": 31,
                  "user_id": "1",
                },
                Object {
                  "date": "2022-04-15",
                  "description": Array [
                    Object {
                      "item_id": 4,
                      "name": "Panqueca de carne",
                      "price": 11,
                      "quantity": 1,
                    },
                  ],
                  "from": "Avenida Conselheiro Rosa e Silva, 1086",
                  "id": "749c347c-4038-418d-b4a9-9e6d22329d19",
                  "orderImage": "http://receitasmais.com.br/wp-content/uploads/2013/04/Panqueca-de-mussarela-e-presunto-1200x675.jpg",
                  "payment": Object {
                    "code": null,
                    "name": null,
                    "security_code": null,
                    "type": "Dinheiro",
                  },
                  "rate": Object {
                    "did": true,
                    "feedback_text": "Descrição",
                    "stars": 3.5,
                  },
                  "restaurant_id": 0,
                  "restaurant_name": "Almir quentinhas",
                  "status": Object {
                    "confirmed": true,
                    "delivering": true,
                    "finished": true,
                    "preparing": true,
                  },
                  "to": "Avenida Jornalista Aníbal Fernandes, 500",
                  "total_price": 11,
                  "user_id": "1",
                },
                Object {
                  "date": "2022-04-13",
                  "description": Array [
                    Object {
                      "item_id": 5,
                      "name": "Panqueca de queijo e presunto",
                      "price": 11,
                      "quantity": 1,
                    },
                  ],
                  "forma_pagamento": Object {
                    "code": null,
                    "name": null,
                    "security_code": null,
                    "type": "Dinheiro",
                  },
                  "from": "Avenida Conselheiro Rosa e Silva, 1086",
                  "id": "296f2c5b-385f-4db9-aca0-1aeb0194d862",
                  "orderImage": "http://receitasmais.com.br/wp-content/uploads/2013/04/Panqueca-de-mussarela-e-presunto-1200x675.jpg",
                  "rate": Object {
                    "did": true,
                    "feedback_text": "Descrição",
                    "stars": 3.5,
                  },
                  "restaurant_id": 0,
                  "restaurant_name": "Almir quentinhas",
                  "status": Object {
                    "confirmed": true,
                    "delivering": true,
                    "finished": true,
                    "preparing": true,
                  },
                  "to": "Avenida Jornalista Aníbal Fernandes, 500",
                  "total_price": 11,
                  "user_id": "1",
                },
              ],
              "error": false,
              "loading": false,
            }
          }
          historyError={[Function]}
          historyRequest={[Function]}
          historySuccess={[Function]}
          postHistory={[Function]}
        >
          <styled.div>
            <div
              className="sc-jqUVSM bxrjtt"
            >
              <styled.div
                className="m-3"
              >
                <div
                  className="sc-breuTD hzEVed m-3"
                >
                  <styled.div>
                    <div
                      className="sc-kDDrLX eKfswp"
                    >
                      Histórico de Pedidos
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-iqcoie iuzcTE"
                    >
                      Filtro de Dias
                      <Styled(styled.div)>
                        <div
                          className="sc-iqcoie sc-crXcEl iuzcTE iKHWLV"
                        >
                          <styled.select
                            onChange={[Function]}
                            value={30}
                          >
                            <select
                              className="sc-evZas iydPvU"
                              onChange={[Function]}
                              value={30}
                            >
                              <styled.option
                                value="30"
                              >
                                <option
                                  className="sc-fnykZs bFkJxx"
                                  value="30"
                                >
                                  30 dias
                                </option>
                              </styled.option>
                              <styled.option
                                value="15"
                              >
                                <option
                                  className="sc-fnykZs bFkJxx"
                                  value="15"
                                >
                                  15 dias
                                </option>
                              </styled.option>
                              <styled.option
                                value="7"
                              >
                                <option
                                  className="sc-fnykZs bFkJxx"
                                  value="7"
                                >
                                  7 dias
                                </option>
                              </styled.option>
                            </select>
                          </styled.select>
                        </div>
                      </Styled(styled.div)>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-ftvSup fTbmAz"
                >
                  <ForwardRef
                    borderless={true}
                  >
                    <table
                      className="table table-borderless"
                    >
                      <tbody>
                        <tr
                          key="10bbd312-47d8-4513-894f-a6556620e763"
                          style={Object {}}
                        >
                          <styled.img
                            photoUrl="https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg"
                          >
                            <img
                              className="sc-hKMtZM dDaXUn"
                            />
                          </styled.img>
                          <td
                            className="p-2"
                          >
                            <styled.div>
                              <div
                                className="sc-papXJ fcJFXi"
                              >
                                <p
                                  style={
                                    Object {
                                      "fontSize": 26,
                                    }
                                  }
                                >
                                  Elias
                                </p>
                                <p>
                                  Ogro Burguer com Batata Média
                                </p>
                                <p
                                  style={
                                    Object {
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  R$ 36,00
                                </p>
                              </div>
                            </styled.div>
                          </td>
                          <td>
                            <Styled(styled.button)
                              disabled={false}
                              onClick={[Function]}
                              variant="blue"
                            >
                              <button
                                className="sc-gsnTZi sc-dkzDqf hFDFfg jjWLoK"
                                disabled={false}
                                onClick={[Function]}
                              >
                                Avaliar Pedido
                              </button>
                            </Styled(styled.button)>
                          </td>
                        </tr>
                        <tr
                          key="749c347c-4038-418d-b4a9-9e6d22329d19"
                          style={Object {}}
                        >
                          <styled.img
                            photoUrl="http://receitasmais.com.br/wp-content/uploads/2013/04/Panqueca-de-mussarela-e-presunto-1200x675.jpg"
                          >
                            <img
                              className="sc-hKMtZM eNIYtn"
                            />
                          </styled.img>
                          <td
                            className="p-2"
                          >
                            <styled.div>
                              <div
                                className="sc-papXJ fcJFXi"
                              >
                                <p
                                  style={
                                    Object {
                                      "fontSize": 26,
                                    }
                                  }
                                >
                                  Almir quentinhas
                                </p>
                                <p>
                                  Panqueca de carne
                                </p>
                                <p
                                  style={
                                    Object {
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  R$ 16,00
                                </p>
                              </div>
                            </styled.div>
                          </td>
                          <td>
                            <Styled(styled.button)
                              disabled={false}
                              onClick={[Function]}
                              variant=""
                            >
                              <button
                                className="sc-gsnTZi sc-dkzDqf kRQeln jjWLoK"
                                disabled={false}
                                onClick={[Function]}
                              >
                                Revisar avaliação
                              </button>
                            </Styled(styled.button)>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </ForwardRef>
                </div>
              </styled.div>
              <styled.div
                className="mt-3"
              >
                <div
                  className="sc-iqcoie iuzcTE mt-3"
                >
                  <Styled(styled.button)
                    key="0"
                    onClick={[Function]}
                    style={
                      Object {
                        "border": "1px solid #ffffff",
                      }
                    }
                  >
                    <button
                      className="sc-gsnTZi sc-ksZaOG kRQeln ituThH"
                      onClick={[Function]}
                      style={
                        Object {
                          "border": "1px solid #ffffff",
                        }
                      }
                    >
                      1
                    </button>
                  </Styled(styled.button)>
                  <Styled(styled.button)
                    key="1"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <button
                      className="sc-gsnTZi sc-ksZaOG kRQeln ituThH"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      2
                    </button>
                  </Styled(styled.button)>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </History>
      </Connect(History)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Home Component Integration Testing Renders loading correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(History)>
        <History
          getHistory={[Function]}
          history={
            Object {
              "data": Array [],
              "error": false,
              "loading": false,
            }
          }
          historyError={[Function]}
          historyRequest={[Function]}
          historySuccess={[Function]}
          postHistory={[Function]}
        >
          <styled.div>
            <div
              className="sc-jqUVSM bxrjtt"
            >
              <styled.div>
                <div
                  className="sc-hAZoDl iccrFX"
                >
                  <h1>
                    Não há pedidos registrados em sua conta
                  </h1>
                  <h2>
                    Volte a página inicial
                  </h2>
                  <Link
                    className="m-2"
                    to="/home"
                  >
                    <a
                      className="m-2"
                      href="/home"
                      onClick={[Function]}
                    >
                      <Button
                        active={false}
                        disabled={false}
                        variant="primary"
                      >
                        <button
                          className="btn btn-primary"
                          disabled={false}
                          type="button"
                        >
                          Voltar
                        </button>
                      </Button>
                    </a>
                  </Link>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </History>
      </Connect(History)>
    </Router>
  </BrowserRouter>
</Provider>
`;
